Source/rendering/texture.cpp

```diff
@@ -1,22 +1,32 @@
 #include "texture.h"
 
 
-Texture::Texture(unsigned char* data, int width, int height, GLuint components) 
-	: mData(data), mWidth(width), mHeight(height), mType(0){
+Texture::Texture(unsigned char* data, int width, int height, int channels, GLuint components)
+	: mData(data), mWidth(width), mHeight(height), mChannels(channels), mFormat(0){
+
+	stbi_set_flip_vertically_on_load(true);
 
 	switch (components) {
 	case 1:
-		mType = GL_RGB;
```

Comment:
Cases 1 and 2 assume input channels are red and red/green, which may not be true. You should instead use an enum and support only 3/4 channels, no other input.